rust/oak/src/lib.rs

```diff
@@ -66,6 +68,23 @@ impl Write for Channel {
     }
 }
 
+/// Read a serialized protobuf request message from a Channel.
+pub fn read_proto_message<T: ::protobuf::Message>(grpc_channel: &mut Channel) -> T {
```

Comment:
could `grpc_channel` be of type `: Read`? You may need to add a generic type `<R>` and a bound `where R : Read` to make that work though.